modules/caddyhttp/reverseproxy/httptransport.go

```diff
@@ -15,375 +15,407 @@
-	ReadBufferSize int `json:"read_buffer_size,omitempty"`
-
-	// The maximum time to wait for next read from backend. Default: no timeout.
-	ReadTimeout caddy.Duration `json:"read_timeout,omitempty"`
-
-	// The maximum time to wait for next write to backend. Default: no timeout.
-	WriteTimeout caddy.Duration `json:"write_timeout,omitempty"`
-
-	// The versions of HTTP to support. As a special case, "h2c"
-	// can be specified to use H2C (HTTP/2 over Cleartext) to the
-	// upstream (this feature is experimental and subject to
-	// change or removal). Default: ["1.1", "2"]
-	Versions []string `json:"versions,omitempty"`
-
-	// The pre-configured underlying HTTP transport.
-	Transport *http.Transport `json:"-"`
-
-	h2cTransport *http2.Transport
+    // TODO: It's possible that other transports (like fastcgi) might be
+    // able to borrow/use at least some of these config fields; if so,
+    // maybe move them into a type called CommonTransport and embed it?
+
+    // Configures the DNS resolver used to resolve the IP address of upstream hostnames.
+    Resolver *UpstreamResolver `json:"resolver,omitempty"`
+
+    // Configures TLS to the upstream. Setting this to an empty struct
+    // is sufficient to enable TLS with reasonable defaults.
+    TLS *TLSConfig `json:"tls,omitempty"`
+
+    // Configures HTTP Keep-Alive (enabled by default). Should only be
+    // necessary if rigorous testing has shown that tuning this helps
+    // improve performance.
+    KeepAlive *KeepAlive `json:"keep_alive,omitempty"`
+
+    // Whether to enable compression to upstream. Default: true
+    Compression *bool `json:"compression,omitempty"`
+
+    // Maximum number of connections per host. Default: 0 (no limit)
+    MaxConnsPerHost int `json:"max_conns_per_host,omitempty"`
+
+    // If non-empty, which PROXY protocol version to send when
+    // connecting to an upstream. Default: off.
+    ProxyProtocol string `json:"proxy_protocol,omitempty"`
+
+    // URL to the server that the HTTP transport will use to proxy
+    // requests to the upstream. See http.Transport.Proxy for
+    // information regarding supported protocols.
+    //
+    // Providing a value to this parameter results in
+    // requests flowing through the reverse_proxy in the following
+    // way:
+    //
+    // User Agent ->
+    //  reverse_proxy ->
+    //  proxy_protocol ->
+    //  transport_proxy_url -> upstream
+    //
+    // Default: off.
+    TransportProxyURL string `json:"transport_proxy_url,omitempty"`
+
+    // How long to wait before timing out trying to connect to
+    // an upstream. Default: `3s`.
+    DialTimeout caddy.Duration `json:"dial_timeout,omitempty"`
+
+    // How long to wait before spawning an RFC 6555 Fast Fallback
+    // connection. A negative value disables this. Default: `300ms`.
+    FallbackDelay caddy.Duration `json:"dial_fallback_delay,omitempty"`
+
+    // How long to wai/Roundtript for reading response headers from server. Default: No timeout.
```

Comment:
Typo/mistake?

Reply:
Unrelated to my changes, but I can see why we'd think it is due to my bad IDE configuration that converted tabs to spaces.

Reply:
:facepalm: lol yeah, that was a typo

Embarrassingly small PR and I've gone and mucked it all up! haha